### DISCUSSION TITLE
How to check the current position time of the music being played？

### DISCUSSION THREAD

**Discussion by nonzhe** [pschatzmann/ESP32-A2DP#400] [Q&A] [answered]
Thank's your libraries!  It's very easy to use!
Now I want to know the current position time of the music played ,
the mask ESP_AVRC_MD_ATTR_PLAYING_TIME not work.

and must use esp_avrc_ct_send_register_notification_cmd function?

**Comment by pschatzmann** [comment, marked as answer]
It is unclear to me if you called set_avrc_metadata_attribute_mask(ESP_AVRC_MD_ATTR_PLAYING_TIME); and registered the callback to get the result (as described in the Readme)?

If this is not working can always use the "Accessing the Sink Data Stream with Callbacks" to calculate the time from the amount data that you received: You're getting 44100 * 2 * 2 bytes / second.

**Reply by nonzhe** [reply]
use “ESP_AVRC_MD_ATTR_PLAYING_TIME”  will return  total time,not the current position.  

I found  in idf examples：
idf\examples\bluetooth\bluedroid\classic_bt\a2dp_sink\main\bt_app_av.c
 
case ESP_AVRC_RN_PLAY_POS_CHANGED:
        ESP_LOGI(BT_AV_TAG, "Play position changed: %d-ms", event_parameter->play_pos);
        bt_av_play_pos_changed();

static void bt_av_play_pos_changed(void)
{
    /* register notification if peer support the event_id */
    if (esp_avrc_rn_evt_bit_mask_operation(ESP_AVRC_BIT_MASK_OP_TEST, &s_avrc_peer_rn_cap,
                                           ESP_AVRC_RN_PLAY_POS_CHANGED)) {
        esp_avrc_ct_send_register_notification_cmd(APP_RC_CT_TL_RN_PLAY_POS_CHANGE,
                                                   ESP_AVRC_RN_PLAY_POS_CHANGED, 10);
    }
}

**Reply by pschatzmann** [reply]
Feel free to extend the API: I suggest to add an additional method set_avrc_rn_evt_cap_mask() similar to set_avrc_metadata_attribute_mask(), to provide a more generic solution.

esp_avrc_tg_set_rn_evt_cap() is executed in BluetoothA2DPSink::av_hdl_stack_evt() and the notifications are  available in BluetoothA2DPSink::av_notify_evt_handler.

If you don't want to contribute, you can create a subclass and overwrite these methods.
In my initial answer, I already gave you a solution that will work w/o any changes.

**Comment by hientv1999** [comment]
Hi, I check the btc_avrc.c and they already implemented the ESP_AVRC_RN_PLAY_POS_CHANGED notification but I still cannot see any event triggered on iPhone. Meanwhile, Android phone can still trigger this event. Do you know how to fix this problem? Thank you

**Comment by pschatzmann** [comment]
No, I guess this is rather a question that you need to address with Apple or Espressif

**Reply by hientv1999** [reply]
Yeah, I think so. By the way, after investigating a bit further, I realize iPhone sends ESP_AVRC_TG_SET_ABSOLUTE_VOLUME_CMD_EVT event instead of ESP_AVRC_RN_PLAY_POS_CHANGED notify event when I change the play position. That's a really weird implementation by Apple.
